dash/include/dash/algorithm/Sort.h

```diff
@@ -845,8 +843,36 @@ void sort(
   array_t g_partition_data(nunits * nunits * 3, dash::BLOCKED, team);
   std::fill(g_partition_data.lbegin(), g_partition_data.lend(), 0);
 
-  auto const min = static_cast<mapped_type>(g_min.get());
-  auto const max = static_cast<mapped_type>(g_max.get());
```

Comment:
I think there is a `barrier` missing before the call of this.
I can't tell if this is the cause of the issue here, but this is my assumption.